src/transformers/modeling_utils.py

```diff
@@ -2433,6 +2466,16 @@ def save_pretrained(
                 token=token,
             )
 
+    @wraps(PushToHubMixin.push_to_hub)
+    def push_to_hub(self, *args, **kwargs):
+        if "tags" not in kwargs:
+            kwargs["tags"] = self._model_tags
+        elif "tags" in kwargs and self._model_tags is not None:
+            logger.warning(
```

Comment:
(nit) I think using `warnings.warn` would be more appropriate here. I usually tend to follow the rule from https://docs.python.org/3/howto/logging.html:

> [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning

> [logging.warning()](https://docs.python.org/3/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted

Here the problem is most likely avoidable by the user.

Reply:
Thanks! I think that transformers fully switched to using `logger.warning` (I got told that in a PR review from @LysandreJik but I don't remember which one 😢 ) let me try to find that and link the comment here

Reply:
No you can keep using `warnings.warn`! see https://github.com/huggingface/transformers/pull/26527